### PR TITLE
[AOT Refactor] jit compile runtime wrappers

### PR DESCRIPTION
**Pull request by jon-chuang**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #114643
* __->__ #114564
* #114563
* #114562
* #114561
* #114559
* #114558
* #114557
* #114556
* #114555
* #114554
* #114553
* #114552
* #114551
* #114550

---

Part _ of https://github.com/pytorch/pytorch/issues/114548

Total reduction in lines: 5200 lines -> 1100 lines

### CODE REVIEW

File: torch/_logging/_registrations.py [hunk 1]
```diff
@@ -4,7 +4,7 @@
 DYNAMIC = ["torch.fx.experimental.symbolic_shapes", "torch.fx.experimental.sym_node"]
 
 register_log("dynamo", ["torch._dynamo", *DYNAMIC])
-register_log("aot", "torch._functorch.aot_autograd")
+register_log("aot", ["torch._functorch.aot_autograd", "torch._functorch._aot_autograd"])
```
Comment (bdhirsh) [inline torch/_logging/_registrations.py:7]: Does this need to be `torch._functorch._aot_autograd.jit_compile_runtime_wrappers`? (I'm not sure, but have you confirmed that `TORCH_LOGS="aot"` still does the right thing after this PR?)

**Comment by bdhirsh** [comment]
@jon-chuang there are two PR's that landed yesterday that you'll probably have to rebase on top of (sorry about that - one of them was a hotfix to fix a regression):

https://github.com/pytorch/pytorch/pull/114646
https://github.com/pytorch/pytorch/pull/111554

I'll stamp this PR and everything below it, and after the rebase we can land them all in one go. Then we can land the top two PR's (mildly BC-breaking but hopefully in a way that shouldn't affect anyone) afterwards.

**Review by bdhirsh — APPROVED** [review]

**Comment by jon-chuang** [comment]
> I also have https://github.com/pytorch/pytorch/pull/112906 (needed for fp8), I'll hold off on landing it til this refactors merges.

Thanks! Yes, I've successfully rebased on them (hopefully). At least all of the aotdispatch tests seem to be passing.

**Pull request closed without merging**